ansible_base/authentication/authenticator_plugins/saml.py

```
@@ -291,7 +291,16 @@ def extra_data(self, user, backend, response, *args, **kwargs):
 
         # Move group spec up a level if present
         if "Group" in attrs:
+            logger.debug("Setting Group from attribute: Group")
             response["Group"] = attrs["Group"]
+        else:
+            # get configured group attribute set Group from that attribute
+            configuration = self.database_instance.configuration
+            idp_groups_attribute_name = self.configuration_class.settings_to_enabled_idps_fields['IDP_GROUPS']
```

I fell like we should do one of two things here:
1. Swap this if/else? It seems like if the customer has specified something in the configuration for the groups name we should go after that first and then fall back to the default "Group" option if the specified value is not defined.
2. Get the default "Groups" and then append the groups from the user specified groups.
Also, should we update the docs around this field to indicate what the logic is doing?

---

I like the option 1. I intentionally left the code as is at first because I wasn't sure what the intention was. I will check documentation on it to see if there is any mention of "Group" in there.

---

Addressed the comment, re-ordered the logic and attempting to get the groups from configured attribute first, only then falling back to hard-coded `Group` attribute.
As to the documentation, it could use some updates because all it says is:
> Enter user details in the Groups, User Email, Username, User Last Name and User First Name.